.github/workflows/ruff.yaml

```diff
@@ -41,6 +41,14 @@ jobs:
         with:
           name: wheels
           path: dist
+      - name: "Rename release binary"
+        run: |
+          mv target/release/ruff target/release/ruff-macos-x86_64
+      - name: "Upload binary"
+        uses: actions/upload-artifact@v3
+        with:
+          name: bin-macos-x86_64
+          path: target/release/ruff-*
```

Comment:
@messense if you have any pointers how to properly name the release binary, let me know

Reply:
Upload the binary directly will lose its file permission: https://github.com/actions/upload-artifact#permission-loss

I'd create an `tar.gz`/`zip` archive for each platform, and please also include a `sha256` checksum file. For reference: https://github.com/rust-cross/cargo-zigbuild/blob/a079e8e11235064a9ee27d44ef21128fd5a23ce2/.github/workflows/Release.yml#L34-L44